src/DynamoPackages/PackageManagerSearchElement.cs

```diff
@@ -308,45 +309,50 @@ private List<VersionInformation> TransformVersionsToVersionInformation(Greg.Resp
+                // 2. If we are not under host, we care about Dynamo compatibility first and foremost. However, we use fallback to Host compatibility if no Dynamo compatibility is found. In this case, we mark as Incompatible
+                // 3. Only if no compatibility information is found, then we mark as 'Unknown Compatibility'
                 if (!string.IsNullOrEmpty(host))
                 {
-                    // Attempt to retrieve compatibility specific to the host
-                    compatibility = compatibilityMatrix.FirstOrDefault(c => c.name.ToLowerInvariant() == host.ToLowerInvariant());
-
-                    // If host compatibility is missing, fallback to Dynamo compatibility
-                    if (compatibility == null)
-                    {
-                        compatibility = compatibilityMatrix.FirstOrDefault(c => c.name.ToLowerInvariant() == "dynamo");
-                    }
+                    // Check for host-specific compatibility
+                    compatibility = compatibilityMatrix.FirstOrDefault(c => c.name.ToLowerInvariant() == host.ToLowerInvariant())
+                                    ?? compatibilityMatrix.FirstOrDefault(c => c.name.ToLowerInvariant() == "dynamo");
                 }
                 else
                 {
-                    // No host specified, so we only check for Dynamo compatibility
+                    // No host specified (DynamoCore only)
+                    // Check Dynamo compatibility first; if missing, check for host compatibility
                     compatibility = compatibilityMatrix.FirstOrDefault(c => c.name.ToLowerInvariant() == "dynamo");
+
+                    if (compatibility == null)
+                    {
+                        // No Dynamo compatibility, fallback to any host compatibility if present
+                        var hostCompatibility = compatibilityMatrix.FirstOrDefault(c => c.name.ToLowerInvariant() != "dynamo");
+                        if (hostCompatibility != null)
+                        {
+                            // Mark as incompatible if host compatibility is present without Dynamo info
+                            return false;
+                        }
+                    }
                 }
 
-                // If no relevant compatibility info is found, return indeterminate
+                // If no compatibility information is found for both Dynamo and host, return unknown (null)
                 if (compatibility == null)
                 {
                     return null;
                 }
 
-                // Step 2: Check compatibility based on min/max ranges or specific versions
+                // Check compatibility based on min/max ranges or specific versions
                 var versionToCheck = (compatibility.name.ToLowerInvariant() == "dynamo") ? dynamoVersion : hostVersion;
                 return versionToCheck != null && IsVersionCompatible(compatibility, versionToCheck);
```

Comment:
Can you log the exception using dynamo logger instead of console.writeline.

Reply:
Yes, you are right. I left that as a last-resort try/catch, but let's log it correctly.

Reply:
Actually, in here we really don't have access to a DynamoModel instance to get to the Logger .. . Is there a static way to call the logger? Otherwise, I will have to remove the try/catch.

Reply:
Yeah we don't have a static logger object to use. So let's keep it like this, we do need the try catch block incase something fails in that.